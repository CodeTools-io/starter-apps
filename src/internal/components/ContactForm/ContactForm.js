import React from 'react';
import { Box, Button, Form, FormField, TextArea, TextInput } from 'grommet';

export default function ContactForm({ children, ...props }) {
  return (
    <Box className="ContactForm" {...props}>
      <Form>
        <Box gap="medium" fill="horizontal">
          <FormField
            component={TextInput}
            name="name"
            label="Name"
            required={true}
          />
          <FormField
            component={TextInput}
            name="email"
            label="Email"
            required={true}
          />
          <FormField
            component={TextInput}
            name="subject"
            label="Subject"
            required={true}
          />
          <FormField
            component={TextArea}
            name="subject"
            label="Message"
            required={true}
            rows="10"
            fill
          />
          <Box align="end">
            <Button label="Send Message" type="submit" primary />
          </Box>
        </Box>
      </Form>
    </Box>
  );
}
