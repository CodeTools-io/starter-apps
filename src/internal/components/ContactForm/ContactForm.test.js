import React from 'react'
import ContactForm from './ContactForm'

describe('ContactForm', () => {
  test('can render', () => {
    const { container } = renderComponent(
      <ContactForm>
        <p>Content for the ContactForm</p>
      </ContactForm>
    )

    expect(container).toMatchSnapshot()
  })
})

