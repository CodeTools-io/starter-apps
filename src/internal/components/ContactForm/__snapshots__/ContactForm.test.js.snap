// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactForm can render 1`] = `
<div>
  <div
    class="StyledBox-sc-13pk1d4-0 dYebPD ContactForm"
  >
    <form>
      <div
        class="StyledBox-sc-13pk1d4-0 dvrejE"
      >
        <div
          class="StyledBox-sc-13pk1d4-0 dLXLoO FormField__FormFieldBox-m9hood-0 jGEMXj"
        >
          <label
            class="StyledText-sc-1sadyjn-0 drTEXR"
          >
            Name
          </label>
          <div
            class="StyledBox-sc-13pk1d4-0 gmReJc FormField__FormFieldContentBox-m9hood-1"
          >
            <div
              class="StyledTextInput__StyledTextInputContainer-sc-1x30a0s-1 dpaJQp"
            >
              <input
                autocomplete="off"
                class="StyledTextInput-sc-1x30a0s-0 fXMpul"
                name="name"
                value=""
              />
            </div>
          </div>
        </div>
        <div
          class="StyledBox__StyledBoxGap-sc-13pk1d4-1 kDNLiy"
        />
        <div
          class="StyledBox-sc-13pk1d4-0 dLXLoO FormField__FormFieldBox-m9hood-0 jGEMXj"
        >
          <label
            class="StyledText-sc-1sadyjn-0 drTEXR"
          >
            Email
          </label>
          <div
            class="StyledBox-sc-13pk1d4-0 gmReJc FormField__FormFieldContentBox-m9hood-1"
          >
            <div
              class="StyledTextInput__StyledTextInputContainer-sc-1x30a0s-1 dpaJQp"
            >
              <input
                autocomplete="off"
                class="StyledTextInput-sc-1x30a0s-0 fXMpul"
                name="email"
                value=""
              />
            </div>
          </div>
        </div>
        <div
          class="StyledBox__StyledBoxGap-sc-13pk1d4-1 kDNLiy"
        />
        <div
          class="StyledBox-sc-13pk1d4-0 dLXLoO FormField__FormFieldBox-m9hood-0 jGEMXj"
        >
          <label
            class="StyledText-sc-1sadyjn-0 drTEXR"
          >
            Subject
          </label>
          <div
            class="StyledBox-sc-13pk1d4-0 gmReJc FormField__FormFieldContentBox-m9hood-1"
          >
            <div
              class="StyledTextInput__StyledTextInputContainer-sc-1x30a0s-1 dpaJQp"
            >
              <input
                autocomplete="off"
                class="StyledTextInput-sc-1x30a0s-0 fXMpul"
                name="subject"
                value=""
              />
            </div>
          </div>
        </div>
        <div
          class="StyledBox__StyledBoxGap-sc-13pk1d4-1 kDNLiy"
        />
        <div
          class="StyledBox-sc-13pk1d4-0 dLXLoO FormField__FormFieldBox-m9hood-0 jGEMXj"
        >
          <label
            class="StyledText-sc-1sadyjn-0 drTEXR"
          >
            Message
          </label>
          <div
            class="StyledBox-sc-13pk1d4-0 gmReJc FormField__FormFieldContentBox-m9hood-1"
          >
            <textarea
              class="StyledTextArea-sc-17i3mwp-0 dlUnNJ"
              name="subject"
              rows="10"
            />
          </div>
        </div>
        <div
          class="StyledBox__StyledBoxGap-sc-13pk1d4-1 kDNLiy"
        />
        <div
          class="StyledBox-sc-13pk1d4-0 giepwR"
        >
          <button
            class="StyledButton-sc-323bzc-0 hmOcFG"
            type="submit"
          >
            Send Message
          </button>
        </div>
      </div>
    </form>
  </div>
</div>
`;
